examples/flutter_gallery/lib/demo/material/reorderable_list_demo.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+enum _ReorderableListType {
+  /// A list tile that contains a [CircleAvatar].
+  horizontalAvatar,
+
+  /// A list tile that contains a [CircleAvatar].
+  verticalAvatar,
+
+  /// A list tile that contains three lines of text and a checkbox.
+  threeLine,
+}
+
+class ReorderableListDemo extends StatefulWidget {
+  const ReorderableListDemo({ Key key }) : super(key: key);
+
+  static const String routeName = '/material/reorderable-list';
+
+  @override
+  _ListDemoState createState() => new _ListDemoState();
+}
+
+class _ListItem {
+  _ListItem(this.value, this.checkState);
+
+  final String value;
+
+  bool checkState;
+}
+
+class _ListDemoState extends State<ReorderableListDemo> {
+  static final GlobalKey<ScaffoldState> scaffoldKey = new GlobalKey<ScaffoldState>();
+
+  PersistentBottomSheetController<Null> _bottomSheet;
+  _ReorderableListType _itemType = _ReorderableListType.threeLine;
+  bool _reverseSort = false;
+  final List<_ListItem> _items = <String>[
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
+  ].map((String item) => new _ListItem(item, false)).toList();
+
+  void changeItemType(_ReorderableListType type) {
+    setState(() {
+      _itemType = type;
+    });
+    _bottomSheet?.setState(() { });
```

Comment:
These two lines are confusing. If _bottomSheet can be null, then don't both references need `?.`. And why rebuild the bottom sheet if we're going to close it? A comment would be helpful here.

Reply:
Correct.  I've updated and added comments.